### ISSUE TITLE
Ubuntu bug report 589574: libtidy creates invalid tags

### ISSUE THREAD

**Issue by vielmetti** [htacg/tidy-html5#256]
The bug report is at 

https://bugs.launchpad.net/ubuntu/+source/tidy/+bug/589574

with a test case against an old version of libtidy. The diagnosis is "use after free".

Suggestions:
- [ ] replicate the bug on current distributions
- [ ] add test case to tests
- [ ] show that the bug has been fixed in tidy-html5
- [ ] recommend that Ubuntu pick up tidy-html5 to replace the old libtidy.

**Comment by geoffmcl** [comment]
@vielmetti, Wow, WOW, loving the new enthusiam over Tidy! Thanks...

Very much appreciated... keep it up...

And also real glad you **did** recognise this is about an `old version of libtidy` ;=))

Any bug that mentions `libtidy-0.99` means it is from the sourceforge cvs site... and indicates a date circa, before, after, around, 2009, like 20051018 plus, can **not** be accepted as a **Tidy 5.1.9 bug**! 

Read the **long** history... the efforts to update the sf cvs site, etc, etc, etc...

Of course the **ONLY** suggestion is `recommend that Ubuntu pick up tidy-html5 to replace the old libtidy`... **no other choice**! ;=))

I am sure you will understand that only **bug** against `libTidy 5.1.9` plus can be accepted here...

At this point, unless there is a very compelling reason, will **not** back port any current fixes to release 5.0.0, making a 5.0.1 release neccessary... 

Would prefer we look forward to a `release 5.2`, perhaps in the not so distant future...

And just to be clear I try hard to leave no stone unturned...

An input of:

```
<!DOCTYPE html PUBLIC "-//W3C//DTD XHTML 1.0 Transitional//EN" "http://www.w3.org/TR/xhtml1/DTD/xhtml1-transitional.dtd">
<html><body>
<p><b><i><a href="A"> <big>B</i></b>C</a></p>
</body></html>
```

Output:

```
line 2 column 7 - Warning: <body> isn't allowed in <head> elements
line 2 column 7 - Info: <head> previously mentioned
line 3 column 23 - Warning: replacing unexpected i by </i>
line 3 column 29 - Warning: discarding unexpected </i>
line 3 column 33 - Warning: inserting implicit <big>
line 3 column 33 - Warning: replacing unexpected b by </b>
line 3 column 33 - Warning: discarding unexpected </b>
line 3 column 37 - Warning: inserting implicit <?>
line 3 column 37 - Warning: missing </?> before </a>
line 3 column 7 - Warning: missing </i> before </p>
line 3 column 4 - Warning: missing </b> before </p>
line 2 column 7 - Warning: inserting missing 'title' element
line 3 column 33 - Warning: trimming empty <big>
Info: Doctype given is "-//W3C//DTD XHTML 1.0 Transitional//EN"
Info: Document content looks like XHTML 1.0 Strict
12 warnings, 0 errors were found!

<!DOCTYPE html PUBLIC "-//W3C//DTD XHTML 1.0 Strict//EN"
    "http://www.w3.org/TR/xhtml1/DTD/xhtml1-strict.dtd">
<html xmlns="http://www.w3.org/1999/xhtml">
<head>
<meta name="generator" content=
"HTML Tidy for HTML5 for Windows version 5.1.9" />
<title></title>
</head>
<body>
<p><b><i><a href="A"><big>B</big><?>C</?></a></i></b></p>
</body>
</html>
```

And can **not** see the `invalid tags change on each run` suggested, indicating that it is `fixed`!

A simple test you could have done before wasting time on this issue...

Closed!

**Comment by vielmetti** [comment]
Thanks Geoff. I updated the Ubuntu bug to mark as "fixed in tidy-html5 5.0.0" - not a waste of time, I hope, because it's useful to have a stable URL to reference when marking a bug as closed.

**Comment by geoffmcl** [comment]
@vielmetti, thanks for updating the Ubuntu bug information, I think ;=))

I guess my point was more that I do not feel this `issues` is the right place to post this now **ancient not now a bug** report...

And please, please avoid reviewing, re-opening the some 142 **open** bugs on the sourceforge tidy bugs list, and other tidy lists on sf! As mentioned I/We (htacg) offered to update the sf source, review, test, comment, and close them **all**...

Still awaiting a response for the tidy sf maintainers ;=((

But if you do have `others` you want to comment on, maybe they could be added to a list like - https://lists.w3.org/Archives/Public/html-tidy/ - or other lists, tidy and/or Ubuntu... 

And on reading this type of report may have just replied `tks`...
